client/components/domain-diagnostic/index.jsx

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+/**
+ * Internal dependencies
+ */
+import Notice from 'components/notice';
+import NoticeAction from 'components/notice/notice-action';
+
+export default React.createClass( {
+
+	imageChecker: null,
+
+	displayName: 'DomainDiagnostic',
+
+	propTypes: {
+		domain: React.PropTypes.string.isRequired,
+	},
+	getInitialState: function() {
+		return {
+			domainDidFail: false
+		};
+	},
+
+	componentDidMount() {
+		const self = this;
+		this.imageChecker = new Image();
+		this.imageChecker.onerror = function() {
+			if ( self.isMounted() ) {
+				self.setState( { domainDidFail: true } );
+			}
+		}
+		this.imageChecker.src = 'http://' + this.props.domain + '/w32.gif';
```

Comment:
:smile: While, I do appreciate the simple hackery of this client side check, I think we'd be better off with a proper server side endpoint.